File: commands/create.js
// COLORS
import { bold, green, gray } from 'https://deno.land/std@0.83.0/fmt/colors.ts'

//JSON
import { writeJson } from 'https://deno.land/x/jsonfile/mod.ts';

export const createLocalTask = async (name, json, HOME) => {
    const description = prompt(`${green(bold('?'))} Please give your task a description:`)
    const date = prompt(`${green(bold('?'))} When should this task finish:`)
    const favourite = prompt(`${green(bold('?'))} Is this task your favourite? ${gray('(yes / no)')}`)
    
    const task = {
        name: name,
        description: description,
        date: date,
        favourite: favourite,
        finish: false
    }

    if (!task.name) {
        task.name = 'my Task'
    }

    json.tasks.push(task)

    const data = JSON.stringify(json)

    // WRITE
    Deno.writeTextFileSync(`${HOME}/tascli.json`, data)


    Deno.exit(200)
}

File: lib/commands/create.dart
import 'dart:io';
import 'package:prompts/prompts.dart' as prompts;
import 'package:tascli/helpers/error.dart';
import 'package:tascli/helpers/json.dart';

void createLocalTask(Map<String, dynamic> json, String taskName, String taskDescription, String taskFavourite) {
  Map<String, dynamic> task = new Map();

  if (taskName == null || taskDescription == null) {
    missingArgument();
    exit(400);
  }


  task['name'] = taskName;
  task['description'] = taskDescription;
  task['favourite'] = taskFavourite != null ? taskFavourite : false;
  task['state'] = json['states'][0]['name'];


  List<dynamic> tasks = json['tasks'];
  tasks.add(task);
  writeJson(json);

  print(json['tasks'].indexOf(task));

  task['id'] = json['tasks'].indexOf(task);
  task['id'] = task['id'] + 1;
  writeJson(json);
}
